lang/python39/files/patch-threadid-older-systems.diff

```diff
@@ -0,0 +1,22 @@
+diff --git Python/thread_pthread.h Python/thread_pthread.h
+index e6910b3..ff9bb1f 100644
+--- Python/thread_pthread.h
++++ Python/thread_pthread.h
+@@ -331,7 +331,17 @@ PyThread_get_thread_native_id(void)
+         PyThread_init_thread();
+ #ifdef __APPLE__
+     uint64_t native_id;
++#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
++    native_id = pthread_mach_thread_np(pthread_self());
++#elif MAC_OS_X_VERSION_MIN_REQUIRED < 1060
++    if (&pthread_threadid_np) {
```

Comment:
Should explicitly check `!= NULL` here.

Reply:
You are exactly right. I pinched the guts of this patch from Jeremy <https://trac.macports.org/browser/trunk/dports/x11/mesa/files/1001-darwin-Address-a-build-failure-on-Leopard-and-earlie.patch?rev=93271> but it should check against NULL.

Now -- to think for a moment -- is there any problem with NULL being defined everywhere in all the needed compilers?

Reply:
I decided not, because NULL is used later on in this patch already.